server/manager/src/main/java/org/apache/accumulo/master/TabletGroupWatcher.java

```diff
@@ -244,6 +246,7 @@ public void run() {
             }
```

Comment:
With this design, the "Always follow through with assignments" check would not need to exist, they would have this logic built into the types. For example, ```AssignedTabletHostedGoal``` is the only class that has a current state of ```ASSIGNED```. 

I am not sure what this second check is trying to do... something with ordering during shutdown of the whole cluster?